gradio/components.py

```diff
@@ -462,7 +467,7 @@ def preprocess(self, x: float | None) -> Optional[float]:
         """
         if x is None:
             return None
-        return float(x)
+        return int(x) if self.integer else float(x)
```

Comment:
Let's update the docstring and type hint.

Reply:
Done!